drivers/serial/uart_rtt.c

```diff
@@ -71,12 +71,13 @@ static int uart_rtt_poll_in(const struct device *dev, unsigned char *c)
  * @param dev UART device struct
  * @param c Character to send
  */
-static void uart_rtt_poll_out(const struct device *dev, unsigned char c)
+static int uart_rtt_poll_out(const struct device *dev, unsigned char c)
 {
 	const struct uart_rtt_config *config = dev->config;
 	unsigned int ch = config ? config->channel : 0;
+	unsigned int ret = SEGGER_RTT_Write(ch, &c, 1);
 
-	SEGGER_RTT_Write(ch, &c, 1);
+	return ret > 0 ? 0 : -EIO;
```

Comment:
`>=`? RTT can have a buffer depending upon Kconfig settings so I don't think it's right for 0 to return an error here

Reply:
As far as I understood, 0 means that the character was dropped and not written into the buffer, which should be indicated with an error, no?

Reply:
https://wiki.segger.com/RTT#SEGGER_RTT_Write.28.29

> \>= 0 | Number of bytes which have been sent.
> \< 0 | Error.

Reply:
Interesting. How can an `unsigned` function return a negative error code?

In any case: If I want to send 1 byte, but 0 bytes "have been sent", that's an error in my understanding.